sysfs-led-white.c

```diff
@@ -0,0 +1,204 @@
+typedef struct
+{
+  int maxval;
+  int fd_val;
+} led_channel_white_t;
+
+/* ------------------------------------------------------------------------- *
+ * ONE_CHANNEL
+ * ------------------------------------------------------------------------- */
+
+static void led_channel_white_init       (led_channel_white_t *self);
+static void led_channel_white_close      (led_channel_white_t *self);
+static bool led_channel_white_probe      (led_channel_white_t *self, const led_paths_white_t *path);
+static void led_channel_white_set_value  (const led_channel_white_t *self, int value);
+
+/* ------------------------------------------------------------------------- *
+ * ALL_CHANNELS
+ * ------------------------------------------------------------------------- */
+
+static void led_control_white_map_color  (int r, int g, int b, int *white);
+static void led_control_white_value_cb   (void *data, int r, int g, int b);
+static void led_control_white_close_cb   (void *data);
+
+bool        led_control_white_probe      (led_control_t *self);
+
+/* ========================================================================= *
+ * ONE_CHANNEL
+ * ========================================================================= */
+
+static void
+led_channel_white_init(led_channel_white_t *self)
+{
+  self->maxval   = -1;
+  self->fd_val   = -1;
+}
+
+static void
+led_channel_white_close(led_channel_white_t *self)
+{
+  led_util_close_file(&self->fd_val);
+}
+
+static bool
+led_channel_white_probe(led_channel_white_t *self,
+                        const led_paths_white_t *path)
+{
+  bool res = false;
+
+  led_channel_white_close(self);
+
+  if( (self->maxval = led_util_read_number(path->max)) <= 0 )
+  {
+    goto cleanup;
+  }
+
+  if( !led_util_open_file(&self->fd_val,   path->val) )
+  {
+    goto cleanup;
+  }
+
+  res = true;
+
+cleanup:
+
+  if( !res ) led_channel_white_close(self);
+
+  return res;
+}
+
+static void
+led_channel_white_set_value(const led_channel_white_t *self, int value)
+{
+  if( self->fd_val != -1 )
+  {
+    dprintf(self->fd_val, "%d", led_util_scale_value(value, self->maxval));
+  }
+}
+
+/* ========================================================================= *
+ * ALL_CHANNELS
+ * ========================================================================= */
+
+static void
+led_control_white_map_color(int r, int g, int b, int *white)
+{
+  /* Use maximum value from requested RGB value */
+  if( r < g ) r = g;
+  if( r < b ) r = b;
+  *white = r;
+}
+
+static void
+led_control_white_value_cb(void *data, int r, int g, int b)
+{
+  const led_channel_white_t *channel = data;
+
+  int white = 0;
+  led_control_white_map_color(r, g, b, &white);
+
+  led_channel_white_set_value(channel + 0, white);
+}
+
+static void
+led_control_white_close_cb(void *data)
+{
+  led_channel_white_t *channel = data;
+  led_channel_white_close(channel + 0);
+}
+
+bool
+led_control_white_probe(led_control_t *self)
+{
+  /** Sysfs control paths for Red + Green leds */
```

Comment:
The "white" backend is basically reduced from the "redgreen" one. Seems I missed some refs to the original. Will fix.